webapp/src/ts/reducers/contacts.ts

```diff
@@ -81,12 +81,90 @@ const removeContact = (state, contact) => {
   return { ...state, contacts, contactsById };
 };
 
+const setLoadingSelectedContact = (state) => {
+  return { ...state, loadingSelectedChildren: true, loadingSelectedReports: true };
+};
+
+const setContactsLoadingSummary = (state, value) => {
+  return { ...state, loadingSummary: value };
+};
+
+const receiveSelectedContactChildren = (state, children) => {
+  return {
+    ...state,
+    loadingSelectedChildren: false,
+    selected: Object.assign({}, state.selected, { children }),
+  };
+};
+
+const receiveSelectedContactReports = (state, reports) => {
+  return {
+    ...state,
+    loadingSelectedReports: false,
+    selected: Object.assign({}, state.selected, { reports }),
+  };
+};
+
+const setSelectedContact = (state, selected) => {
+  return { ...state, selected };
+};
+
+const updateSelectedContact = (state, summary) => {
```

Comment:
@njogz I thought I can use this one for updated any contact property, but it's limited to `summary`. I think better 
Change the name to `updateSelectedContactSummary` 
or 
Change the function to update any property from contact: `selected: { ...state.selected, ...newSelected }`